homework/App.js

```diff
@@ -15,15 +15,38 @@ class App {
     // Add code here to initialize your app
     // 1. Create the fixed HTML elements of your page
     // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
     const root = document.getElementById('root');
-
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
-
+    // TODO: replace with your own code
+    const header = Util.createAndAppend('div', root, { id: 'header' });
+    Util.createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const select = Util.createAndAppend('select', header, {
+      id: 'select',
+      'aria-label': 'HYF Repositories',
+    });
+    select.addEventListener('change', () => this.fetchContributorsAndRender(select.value));
+    Util.createAndAppend('div', root, {
+      id: 'container',
+    });
     try {
-      const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // TODO: add your own code here
+      const response = await fetch(url);
```

Comment:
I will also be a bit lenient here but you were actually supposed to use the `fetchJSON` method inside the Utils class to fetch repository and contributors information. 😉

Reply:
:smile:  Yes. The reason that I have used fetch is, Joost told us in the previous lecture to use fetch, so I thought that I have to use fetch here also.